pkg/streamingpromql/types/data.go

```diff
@@ -186,3 +186,9 @@ func NewRangeQueryTimeRange(start time.Time, end time.Time, interval time.Durati
 		StepCount:  int((endT-startT)/intervalMs) + 1,
 	}
 }
+
+// PointIdx returns the index in the QueryTimeRange that the timestamp, t, falls on.
+// t must be in line with IntervalMs (ie the step).
+func (q *QueryTimeRange) PointIdx(t int64) int64 {
```

Comment:
[nit] How about `PointIndex`? I don't think there's a need to abbreviate "index" here.

(although, in fairness, then we should probably change `IntervalMs` to `IntervalMilliseconds`)